File: python/zj-b231.py
#!/usr/bin/env python 

while True:
    try:
        n = int(input())
        lst = []
        for _ in range(n):
            p, b = map(int, input().split())
            lst.append((b, p))

        lst.sort(key = lambda x: (x[0], x[1]))
        cpt = 0 #累計花費的印刷時間
        ans = 0
        while len(lst):
            tp = lst.pop()
            cpt += tp[1]
            ans = max(ans, cpt + tp[0])
        print(ans)
        print()
        _ = input() #兩筆測資之間有一空白列
    except:
        break

File: cpp/zj-b231.cpp

#include <iostream>
#include <queue>
using namespace std;

int main(){
    int n, p, b;
    priority_queue <pair<int, int> > pq;
    while (cin >> n){
        for (int i = 0; i < n; i++){
            cin >> p >> b;
            pq.push(make_pair(b, p));
        }
        int cpt = 0; //累計花費的印刷時間
        int ans = 0;
        while (!pq.empty()){
            cpt += pq.top().second;
            ans = max(ans, cpt + pq.top().first);
            pq.pop();
        }
        cout << ans << "\n\n";
    }
    return 0;
}
